### DISCUSSION TITLE
FormDialog 如何提交数据后再关闭

### DISCUSSION THREAD

**Discussion by ctforcp** [alibaba/formily#1303] [Q&A 2.x] [answered]
[https://codesandbox.io/s/hardcore-germain-uyhmk?file=/App.tsx](https://codesandbox.io/s/hardcore-germain-uyhmk?file=/App.tsx)

> 如题，想让弹出后 FormDialog ，等数据提交到后台再关闭，而不是点确定就直接关闭

**Comment by janryWang** [comment, marked as answer]
自己定制一下footer，用Submit组件来提交，在onSubmit处理请求，请求结束之后调用resolve

**Reply by ctforcp** [reply]
定制footer，用Submit组件会报错 ，外面再添加Form组件照样报错。

<https://codesandbox.io/s/headless-resonance-dlrme?file=/App.tsx>

![t](https://user-images.githubusercontent.com/49541356/116175520-0eed1d00-a743-11eb-9db3-dbbc69a8374b.gif)

**Reply by janryWang** [reply]
https://codesandbox.io/s/ancient-snow-pqno4?file=/App.tsx 你用法不对，看看这个

**Reply by ctforcp** [reply]
好的 谢谢大佬

**Comment by Nokecy** [comment]
最新版本的FormDialog 还是需要这样操作吗? 有没有简单一点的写法~~ @janryWang

**Reply by janryWang** [reply]
forConfirm呀，已经够简单了

**Reply by Nokecy** [reply]
forConfirm 出现提交异常,loading 状态不消失 ,稍后我给一个复现~~~

**Reply by Nokecy** [reply]
https://codesandbox.io/s/hungry-sara-g5c7c?file=/App.tsx:1170-1295

大佬看看 是我的用法不对吗? @janryWang

**Reply by janryWang** [reply]
提个bug吧，忘记catch错误的问题了

**Reply by AstraeusSEA** [reply]
@formily/element-plus <FormDialog.Footer> 不生效
